extensions/mongodb-client/runtime/src/main/java/io/quarkus/mongodb/runtime/MongoClients.java

```diff
@@ -326,9 +328,41 @@ private MongoClientSettings createMongoConfiguration(MongoClientConfig config) {
             settings.uuidRepresentation(config.uuidRepresentation.get());
         }
 
+        settings = customize(name, settings);
+
         return settings.build();
     }
 
+    private boolean doesNotHaveClientNameQualifier(Bean<?> bean) {
+        for (Annotation qualifier : bean.getQualifiers()) {
+            if (qualifier.annotationType().equals(MongoClientName.class)) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    private MongoClientSettings.Builder customize(String name, MongoClientSettings.Builder settings) {
+        // If there is no customizer for the default client, use the first one that does not have a client name qualifier
+        // Otherwise look for the right customizer
+        // At build time, we check that we have at most one customizer per client.
+        if (MongoClientBeanUtil.isDefault(name)) {
+            var list = customizers.handlesStream().filter(h -> doesNotHaveClientNameQualifier(h.getBean()))
+                    .collect(Collectors.toList());
```

Comment:
You can use findFirst() here to slightly improve readability but it's not mandatory, you can leave it as is.

Reply:
I switched to findFirst.

It's what I was using initially, but I was not happy with the maybe.get().get().customize(....). The first `get` retrieve the value from the optional. The second `get` retrieves the bean instance.

Reply:
Well, I said 'can' so you could have keep it as is ;)